### PR TITLE
Making tensorflow optional in test_keras_surrogate

### PR DESCRIPTION
**Pull request by carldlaird**
## Fixes
- using pyomo attempt_import to make 'tensorflow.keras' an optional import and skipping all of test_keras_surrogate file to skip tests if tensorflow.keras is not available
- removing tensorflow from the list of required dependencies

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/surrogate/keras_surrogate.py [hunk 1]
```diff
@@ -16,7 +16,13 @@
 import numpy as np
 import json
 import os.path
-import tensorflow.keras as keras
+
+try:
+    import tensorflow.keras as keras
+    keras_available = True
+except:
+    keras_available = False
```
Comment (jsiirola) [inline idaes/surrogate/keras_surrogate.py:24]: You aren't testing that keras is available before using it, which means that a user that does not have tensorflow installed will just get a `NameError: name `keras` is not defined`.  I would recommend either adding a test for keras_available (and an appropriate exception), or just use Pyomo's `attempt_import()` which will generate the exception automatically when `keras` is used:
```
from pyomo.common.dependencies import attempt_import
keras, keras_available = attempt_import('tensorflow.keras`)
```

**Review by lbianchi-lbl — APPROVED** [review]
There was a failing pylint check, but using `attempt_import()` might have taken care of that.

File: setup.py [hunk 2]
```diff
@@ -65,7 +65,6 @@ def rglob(path, glob):
         "numpy",
         "networkx",
         "omlt==0.3", # fix the version for now as package evolves
-        "tensorflow",
```
Comment (jsiirola) [inline setup.py:68]: @lbianchi-lbl: I am all for culling required dependencies, but as we do that, should we start an "optional" target with something like:
```
extras_require = {
    'optional': [
        'tensorflow',  # idaes.surrogates.keras_surrogate
    ]
}
```

Comment (lbianchi-lbl) [reply setup.py:68]: Yeah, I think it's a good idea. I thought we could deal with the optional dependencies all at once, but this looks like a good place to start.

File: idaes/surrogate/keras_surrogate.py (hunk 1 above)
Comment (carldlaird) [reply idaes/surrogate/keras_surrogate.py]: I made this change.

File: setup.py (hunk 2 above)
Comment (lbianchi-lbl) [reply setup.py:68]: @carldlaird @jsiirola I can do this as part of #633. I propose we get this merged in as soon as all the tests pass.

File: idaes/surrogate/keras_surrogate.py [hunk 3]
```diff
@@ -16,7 +16,10 @@
 import numpy as np
 import json
 import os.path
-import tensorflow.keras as keras
+
+from pyomo.common.dependencies import attempt_import
+keras, keras_available = attempt_import('tensorflow.keras')
```
Comment (jsiirola) [inline idaes/surrogate/keras_surrogate.py:21]: One more quick change: by default, the error message emitted by `attempt_import` is:
```
The tensorflow.keras module (an optional Pyomo dependency) failed to import: ModuleNotFoundError: No module named 'tensorflow'
```
If we don't want this to blame Pyomo, then we can specify our own message:
```
keras, keras_available = attempt_import(
    'tensorflow.keras', 'The tensorflow.keras module (an optional IDAES dependency) failed to import'
)

**Review by jsiirola — APPROVED** [review]

**Pull request merged**